tests/unit/python/foglamp/services/core/api/test_notification.py

```diff
@@ -400,6 +406,26 @@ def client(self, loop, test_client):
         update_configuration_item_bulk_calls = [call('Test Notification', {'enable': 'false', 'rule': 'threshold', 'description': 'Test Notification', 'channel': 'email', 'notification_type': 'one shot'})]
         update_configuration_item_bulk.assert_has_calls(update_configuration_item_bulk_calls, any_order=True)
 
+    async def test_post_notification_duplicate_name(self, mocker, client):
+        mocker.patch.object(ServiceRegistry, 'get', return_value=mock_registry)
+        mocker.patch.object(notification, '_hit_get_url', return_value=mock_get_url("/foglamp/notification/plugin"))
+        mocker.patch.object(notification, '_hit_post_url',
+                            side_effect=[mock_post_url("/notification/Test Notification"),
+                                         mock_post_url("/notification/Test Notification/rule/threshold"),
+                                         mock_post_url("/notification/Test Notification/delivery/email")])
+        mocker.patch.object(connect, 'get_storage_async')
+        mocker.patch.object(ConfigurationManager, '__init__', return_value=None)
+        mocker.patch.object(ConfigurationManager, '_read_category_val', return_value=mock_read_category_val())
```

Comment:
why `mocker.patch.object(ConfigurationManager, '_read_category_val', return_value=mock_read_category_val())` We can simply avoid this heavy mocking as it will return for bad request too early with checking.